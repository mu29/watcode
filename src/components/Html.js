/* eslint-disable react/no-danger */
import React from 'react'
import { PropTypes } from 'prop-types'
import { Helmet } from 'react-helmet'

const Html = ({ styles, state, children }) => {
  const helmet = Helmet.renderStatic()
  const htmlAttrs = helmet.htmlAttributes.toComponent()
  const bodyAttrs = helmet.bodyAttributes.toComponent()

  return (
    <html lang="ko" { ...htmlAttrs }>
      <head>
        { helmet.title.toComponent() }
        { helmet.meta.toComponent() }
        { helmet.link.toComponent() }
        <meta charSet="utf8" />
        <link href="/style.css" rel="stylesheet" type="text/css" />
        <link href="//spoqa.github.io/spoqa-han-sans/css/SpoqaHanSans-kr.css" rel="stylesheet" type="text/css" />
        <script defer src="https://use.fontawesome.com/releases/v5.0.13/js/all.js" integrity="sha384-xymdQtn1n3lH2wcu0qhcdaOpQwyoarkgLVxC/wZ5q7h9gHtxICrpcaSUfygqZGOe" crossOrigin="anonymous" />
        { styles }
      </head>
      <body { ...bodyAttrs }>
        <main id="app">{ children }</main>
        <script dangerouslySetInnerHTML={{ __html: state }} />
        <script src="/client.js" />
      </body>
    </html>
  )
}

Html.propTypes = {
  styles: PropTypes.node.isRequired,
  state: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
}

export default Html
