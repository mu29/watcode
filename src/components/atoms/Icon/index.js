import React from 'react'
import { PropTypes } from 'prop-types'
import styled from 'styled-components'
import { color, size } from 'services/style'

const Wrapper = styled.div`
  display: contents;
  color: ${color};
  font-size: ${size};
`

const Icon = ({ icon, ...props }) => (
  <Wrapper { ...props }>
    <i className={ `fas fa-${icon}` } />
  </Wrapper>
)

Icon.propTypes = {
  icon: PropTypes.string.isRequired,
  size: PropTypes.number,
  color: PropTypes.string,
}

Icon.defaultProps = {
  size: 16,
  color: 'gray.90',
}

export default Icon
