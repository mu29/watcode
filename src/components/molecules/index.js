export { default as Field } from './Field'
export { default as IconButton } from './IconButton'
export { default as Tab } from './Tab'
