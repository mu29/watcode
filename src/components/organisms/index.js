export { default as ArtworkList } from './ArtworkList'
export { default as Header } from './Header'
export { default as LoginForm } from './LoginForm'
export { default as ReduxField } from './ReduxField'
export { default as RegisterForm } from './RegisterForm'
export { default as SearchBar } from './SearchBar'
