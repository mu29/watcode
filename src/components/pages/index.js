export { default as HomePage } from './HomePage'
export { default as RankingPage } from './RankingPage'
export { default as BookmarkPage } from './BookmarkPage'
