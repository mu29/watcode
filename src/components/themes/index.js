export { default as basicTheme } from './basic'
