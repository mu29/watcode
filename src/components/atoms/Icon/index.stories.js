import React from 'react'
import { storiesOf } from '@storybook/react'
import Icon from '.'

storiesOf('Atoms/Icon', module)
  .add('기본', () => (
    <Icon icon="star" />
  ))
