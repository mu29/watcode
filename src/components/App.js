import React from 'react'
import PropTypes from 'prop-types'
import Helmet from 'react-helmet'
import { injectGlobal, ThemeProvider } from 'styled-components'
import { authorizeActions, readBookmarksActions } from 'store/actions'
import { auth } from 'services/firebase'
import { basicTheme } from './themes'
import Router from './Router'

/* eslint-disable no-unused-expressions */
injectGlobal`
  * {
    box-sizing: border-box;
  }
  body {
    margin: 0;
  }
`

export default class App extends React.Component {
  static propTypes = {
    store: PropTypes.shape({
      dispatch: PropTypes.func,
      getState: PropTypes.func,
    }).isRequired,
  }

  componentWillMount() {
    const { store } = this.props
    auth.onAuthStateChanged((user) => {
      store.dispatch(authorizeActions.request({ user }))
      store.dispatch(readBookmarksActions.request())
    })
  }

  render() {
    const state = this.props.store.getState()
    const isLoggedIn = state.auth.user.email !== undefined
    return (
      <React.Fragment>
        <Helmet titleTemplate="watcode">
          <title>왓코드 - 오늘 볼 만화 추천해줘!</title>
          <meta charSet="UTF-8" />
          <meta name="description" content="" />
          <meta name="viewport" content="width=device-width, initial-scale=1.0" />
          <meta property="og:site_name" content="watcode" />
        </Helmet>
        <ThemeProvider theme={ basicTheme }>
          <Router isLoggedIn={ isLoggedIn } />
        </ThemeProvider>
      </React.Fragment>
    )
  }
}
