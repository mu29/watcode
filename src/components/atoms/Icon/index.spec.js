import React from 'react'
import { shallow } from 'enzyme'
import { basicTheme } from 'components/themes'
import 'jest-styled-components'
import Icon from '.'

const wrap = (props = {}) => shallow(<Icon theme={ basicTheme } { ...props } />).dive()

describe('<Icon />', () => {
  it('정상적으로 렌더링된다', () => {
    const wrapper = wrap({ icon: 'user' })
    expect(wrapper).toMatchSnapshot()
  })
})
