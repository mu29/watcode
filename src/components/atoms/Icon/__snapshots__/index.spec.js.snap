// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> 정상적으로 렌더링된다 1`] = `
.c0 {
  display: contents;
  color: #424242;
  font-size: 1rem;
}

<div
  className="c0"
  color="gray.90"
  size={16}
>
  <i
    className="fas fa-user"
  />
</div>
`;
